patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -108,12 +108,16 @@
 +   @Nullable
 +   @Override
 +   public net.minecraftforge.common.ToolType getHarvestTool(BlockState state) {
++      if (this.harvestTool == null && net.minecraft.item.AxeItem.field_234662_c_.contains(state.func_185904_a()))
```

Comment:
This is a no for a few reasons, 1) Don't make the field public/modifiable for other people. It's not good API design. The better option would be to add a 'is' method in the AxeItem class.
2) Honestly I think you're attacking the wrong target, you should probably go after 'isToolEffective' not getHarvestTool.
Perhaps we should change the default implementation of isToolEffective to `return getHavestTool(state) == null || getHavestTool(state) == tool`

Reply:
1) You are right, adding a static `isEffectiveMaterial(BlockState)` is better than making this field public. I will change that.
2) All blocks that have a material where the axe is effective have not set a harvest tool. So, only adding the check `getHavestTool(state) == null` to `isToolEffective` would not change that and pickaxes would also be an effective tool for logs for example. So, changing the `getHarvestTool` should also be done.

Reply:
2) If that's the case, then you should do your first idea and fill in the vanilla blocks with their tools based on the material.
You can easily iterate all blocks using the registry. So it shouldn't be hard.
Vanilla blocks do not have state based materials so you don't have to do anything special. 
Modded blocks should override isToolEffective, Honestly getHarvestTool should be deleted. It's mainly still existing for JEI. Considering multiple tools may work on the same state the concept of only having one is not valid anymore.

Reply:
Should I use the ForgeRegistries.BLOCKS registry for this?
Should I remove that method in this PR?

Reply:
Sure
No, 1.17 would be the time, we need to talk about it more.